tests/unittests/core/fileio/fileutilstest.cpp

```diff
@@ -0,0 +1,313 @@
+/*
+ * LibrePCB - Professional EDA for everyone!
+ * Copyright (C) 2013 LibrePCB Developers, see AUTHORS.md for contributors.
+ * https://librepcb.org/
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*******************************************************************************
+ *  Includes
+ ******************************************************************************/
+
+#include <gtest/gtest.h>
+#include <librepcb/core/exceptions.h>
+#include <librepcb/core/fileio/filepath.h>
+#include <librepcb/core/fileio/fileutils.h>
+#if defined(Q_OS_WIN32) || defined(Q_OS_WIN64)  // Windows
+#include <fileapi.h>
+#endif
+
+#include <QtCore>
+
+/*******************************************************************************
+ *  Namespace
+ ******************************************************************************/
+namespace librepcb {
+namespace tests {
+
+/*******************************************************************************
+ *  Test Class
+ ******************************************************************************/
+
+class FileUtilsTest : public ::testing::Test {
+protected:
+  FilePath root;
+  FilePath rootFile;  // source for all operations
+  FilePath rootFileHidden;
+
+  FilePath subdir;
+  FilePath subdirFile;
+  FilePath subdirSubdir;
+  FilePath subdirSubdirFile;
+  FilePath subdirSubdirFileHidden;
+
+  FilePath rootFileMissing;
+  FilePath rootFileCopy;
+  FilePath subdirCopy;
+  FilePath subdirCopyFile;
+  FilePath subdirCopySubdir;
+  FilePath subdirCopySubdirFile;
+  FilePath subdirCopySubdirFileHidden;
+
+  QStringList filter;
+
+  static void SetupFile(const FilePath& pth, const QByteArray& content) {
```

Comment:
Well there was two options about capitals:

 * naming conventions of `gtest`
 * naming conventions of `LibrePCB`

Because my test code *inherits* from `gtest`, and there are *overriden* functions from `gtest`, i chosen the `gtest` version because of consistency across entire object. The other way will generate `setupFile` and `SetUp` functions and this could look ugly for someone (inconsistent across object).

It's not problem for me to use name `setupFile` if that will be the way You want. I'm happy that I predicted that this will be commented in PR before I created PR. Those assumptions makes code cleaner and I don't know which selection You will use.

Reply:
Yeah even if the gtest naming convention is different, I prefer to use our own naming convention whenever possible. It avoids propagating the gtest naming conventions to other parts of our code by copy&paste.

But I still think the SetUp/TearDown functions are not needed anyway (see other comment) which would eliminate the inconsistency anyway :grin: